### ISSUE TITLE
Can't download to internal storage

### ISSUE THREAD

**Issue by handgun50** [instaloader/instaloader#259]
I'm using Termux on Android. When I download photos to instaloader folder, it works perfectly. But, if I want to directly download to internal storage, it fails to download 
```
__  __           _       _      _
|  \/  |_ __     / \   __| |_ __(_) __ _ _ __
| |\/| | '__|   / _ \ / _` | '__| |/ _` | '_ \
| |  | | |     / ___ \ (_| | |  | | (_| | | | |
|_|  |_|_|    /_/   \_\__,_|_|  |_|\__,_|_| |_|

zsh: done       echo |
zsh: abort
 ~  cd instaloader
 ~/instaloader   master  instaloader --no-video-thumbnails --no-captions --no-metadata-json --no-videos --dirname-pattern=/sdcard/script/{target} adrianwahyudi17
WARNING: linker: Unsupported flags DT_FLAGS_1=0x8
WARNING: linker: Unsupported flags DT_FLAGS_1=0x8
Stored ID 258764001 for profile adrianwahyudi17.
Warning: Use --login to download HD version of profile pictures.
/sdcard/script/adrianwahyudi17/2018-12-08_11-21-41_UTC_profile_pic.jpg Traceback (most recent call last):
  File "/data/data/com.termux/files/usr/bin/instaloader", line 11, in <module>
    load_entry_point('instaloader==4.2.2', 'console_scripts', 'instaloader')()
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/instaloader/__main__.py", line 412, in main
    storyitem_filter_str=args.storyitem_filter)
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/instaloader/__main__.py", line 186, in _main
    download_stories, fast_update, post_filter, storyitem_filter)
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/instaloader/instaloader.py", line 941, in download_profiles
    self.download_profilepic(profile)
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/instaloader/instaloader.py", line 65, in call
    return func(instaloader, *args, **kwargs)
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/instaloader/instaloader.py", line 356, in download_profilepic
    os.utime(filename, (datetime.now().timestamp(), date_object.timestamp()))
PermissionError: [Errno 1] Operation not permitted
 ✘  ~/instaloader   master 
```

**Comment by Thammus** [comment]
As I just found out reading this [[GUIDE][HOW-TO]Keep timestamp and attributes correct](https://forum.xda-developers.com/android/general/guide-timestamp-attributes-correct-t2960935), there is a bug in Android which causes `setLastModified()` of class `File` to fail on at least all file systems which are emulated by Android using FUSE.
> The implementation of FUSE by Google runs using the root-account. The disadvantage is that the standard user is not capable of changing any attribute including timestamp.

This problem is resolved with Android 8 and later since FUSE got replaced by SDCardFS.

This leads to three potential solutions for your error:
- Use Android 8 or later
- Invoke Instaloader with root privileges
- Remove all occurrences of `os.utime()` from Instaloader

I just installed Termux on Android 8 and tested Instaloader. Downloading to internal storage worked fine without any problems.
